spec/requests/line_items/update_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe "Update LineItems" do
+      subject(:put_request) do
+        sign_in user
+        put line_item_path(line_item), params: params
+      end
+
+      let!(:product) { FactoryBot.create(:product, :with_image, name: 'Ipod', price: 120) }
```

Comment:
the bang! here is not necessary as you are calling the product while creating the line_item. same for user